google-api-client/src/main/java/com/google/api/client/googleapis/util/Utils.java

```diff
@@ -55,6 +67,68 @@ public static HttpTransport getDefaultTransport() {
     static final HttpTransport INSTANCE = new NetHttpTransport();
   }
 
-  private Utils() {
+  /**
+   * Returns the `cert_provider_command` field in context_aware_metadata.json file.
+   *
+   * @param contextAwareMetadata Input stream for ~/.secureConnect/context_aware_metadata.json file.
+   * @return `cert_provider_command` field
+   * @since 1.31
+   */
+  static List<String> extractCertificateProviderCommand(InputStream contextAwareMetadata)
+      throws IOException {
+    JsonParser parser = JsonFactoryInstanceHolder.INSTANCE.createJsonParser(contextAwareMetadata);
+    ContextAwareMetadataJson json = parser.parse(ContextAwareMetadataJson.class);
+    return json.getCommands();
   }
+
+  /**
+   * Returns if mTLS client certificate should be used. mTLS client certificate is used if and only
+   * if "GOOGLE_API_USE_CLIENT_CERTIFICATE" environment variable value is "true".
+   *
+   * @return If mTLS client certificate should be used
+   * @since 1.31
+   */
+  public static Boolean useMtlsClientCertificate() {
+    String useClientCertificate = System.getenv(GOOGLE_API_USE_CLIENT_CERTIFICATE);
+    return "true".equals(useClientCertificate);
+  }
+
+  /**
+   * Returns the default KeyStore for mutual TLS.
+   *
+   * <p>Default client certificate can be generated by running the commands in
+   * ~/.secureConnect/context_aware_metadata.json file. If this json exists, and the default client
+   * certificate exists, this function returns a KeyStore object created with the default client
+   * certificate and the key store password is "". Otherwise, null will be returned.
+   *
+   * @return KeyStore for mutual TLS.
+   * @since 1.31
+   */
+  public static KeyStore loadDefaultMtlsKeyStore() throws IOException, GeneralSecurityException {
+    File file = new File(CONTEXT_AWARE_METADATA_PATH);
+    if (!file.exists()) {
+      return null;
+    }
+
+    // Load the cert provider command from the json file.
+    InputStream stream = new FileInputStream(file);
+    List<String> command = extractCertificateProviderCommand(stream);
+
+    // Call the command.
+    Process process = new ProcessBuilder(command).start();
+    int exitCode = 0;
+    try {
+      exitCode = process.waitFor();
```

Comment:
We should also set a timeout here otherwise this could potentially block the entire process.

Reply:
Timeout is added.